File: src/network/packet/impl/TemplateSyncPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\ClientboundPacket;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\util\PacketData;
use pocketcloud\cloud\template\Template;

final class TemplateSyncPacket extends CloudPacket implements ClientboundPacket {

    public function __construct(
        private readonly ?Template $template = null,
        private readonly bool $removal = false
    ) {}

    public function handle(ServerClient $client): void {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->template, $this->removal);
    }

    public function decodePayload(PacketData $packetData): void {}

    public function getTemplate(): ?Template {
        return $this->template;
    }

    public function isRemoval(): bool {
        return $this->removal;
    }

    public static function create(Template $template, bool $removal): self {
        return new self($template, $removal);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/TemplateSyncPacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.network.packet.ClientboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.network.packet.data.PacketData;
import de.pocketcloud.cloud.template.Template;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class TemplateSyncPacket extends CloudPacket implements ClientboundPacket {

    private Template template;
    private boolean removal;

    public TemplateSyncPacket(Template template, boolean removal) {
        this.template = template;
        this.removal = removal;
    }

    @Override
    public void handle(@NotNull ServerClient client) {}

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(template, removal);
    }

    @Override
    public void decodePayload(PacketData packetData) {}

    public static TemplateSyncPacket create(Template template, boolean removal) {
        return new TemplateSyncPacket(template, removal);
    }
}
